aws/resource_aws_directory_service_directory.go

```diff
@@ -179,6 +179,56 @@ func resourceAwsDirectoryServiceDirectory() *schema.Resource {
 					directoryservice.DirectoryEditionStandard,
 				}, false),
 			},
+			"radius_settings": {
+				Type:     schema.TypeList,
+				MaxItems: 1,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"protocol": {
+							Type:     schema.TypeString,
+							Required: true,
+							ValidateFunc: validation.StringInSlice([]string{
+								directoryservice.RadiusAuthenticationProtocolPap,
+								directoryservice.RadiusAuthenticationProtocolChap,
+								directoryservice.RadiusAuthenticationProtocolMsChapv1,
+								directoryservice.RadiusAuthenticationProtocolMsChapv2,
+							}, false),
+						},
+						"label": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"port": {
+							Type:     schema.TypeInt,
+							Required: true,
+						},
+						"retries": {
+							Type:     schema.TypeInt,
+							Required: true,
+						},
+						"servers": {
+							Type:     schema.TypeSet,
+							Required: true,
+							Elem:     &schema.Schema{Type: schema.TypeString},
```

Comment:
add validation https://docs.aws.amazon.com/directoryservice/latest/devguide/API_RadiusSettings.html

ValidateFunc: validation.IntBetween(0, 256),

Reply:
Doesn't seem possible to add ValidateFunc to sets.  
I get the error:

```
1 error occurred:
        	* resource aws_directory_service_directory: servers: ValidateFunc and
        ValidateDiagFunc are not yet supported on lists or sets.
```

Reply:
it should be on the elem level:

```
Elem:     &schema.Schema{
Type: schema.TypeString,
ValidateFunc: validation.IntBetween(0, 256),
},
```

Reply:
Done, although it's a string, not an int.  Thanks